### PR TITLE
coprocessor/dag/expr: add to_days

### PR DESCRIPTION
**Pull request by GinYM**
## What have you changed? (mandatory)

implement builtin function: ToDays in coprocessor.
[#3275](https://github.com/tikv/tikv/issues/3275)

## What are the type of the changes? (mandatory)

Improvement

## How has this PR been tested? (mandatory)

Unit Test

## Does this PR affect documentation (docs) update? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

### CODE REVIEW

**Comment by siddontang** [comment]
Thanks @GinYM 

Please sign off your all commits. See https://github.com/tikv/tikv/blob/master/CONTRIBUTING.md#signing-off-the-commit

If you have already pushed some commits before, you can use `git rebase -i master --signoff` to resign them off.

File: src/coprocessor/dag/expr/builtin_time.rs [hunk 1]
```diff
@@ -1119,4 +1130,35 @@ mod tests {
             Datum::Time(Time::parse_utc_datetime("0000-00-00 00:00:00.000000", 6).unwrap()),
         );
     }
+
+    #[test]
+    fn test_to_days() {
+        let cases = vec![
```
Comment (rleungx) [inline src/coprocessor/dag/expr/builtin_time.rs]: Can you add more test cases? e.g.`"0000-00-00"`, `"1992-13-00"`, `"2007-10-07 23:59:61"`, `123456789`

Comment (GinYM) [reply src/coprocessor/dag/expr/builtin_time.rs]: It seems that `Time::parse_utc_datetime` cannot check invalid date such as `"1992-13-00"`. Do we need to implement the checking in `pub fn to_days()` or in `parse_datetime_internal()` which is in the file `mod.rs`

File: src/coprocessor/dag/expr/builtin_time.rs [hunk 2]
```diff
@@ -343,6 +343,17 @@ impl ScalarFunc {
     ) -> Result<Option<Cow<'a, Time>>> {
         Ok(Some(Cow::Owned(mysql::time::zero_datetime(ctx.cfg.tz))))
     }
+
+    #[inline]
+    pub fn to_days(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<i64>> {
+        let t: Cow<Time> = try_opt!(self.children[0].eval_time(ctx, row));
+        if t.is_zero() {
```
[the remark this answers is not in the document]
Comment (GinYM) [reply src/coprocessor/dag/expr/builtin_time.rs]: Zero date in MySQL should be invalid. See https://dev.mysql.com/doc/refman/5.7/en/date-and-time-functions.html#function_to-days

[the remark this answers is not in the document]
Comment (huachaohuang) [reply src/coprocessor/dag/expr/builtin_time.rs]: OK, thanks.

File: src/coprocessor/dag/expr/builtin_time.rs (hunk 1 above)
Comment (huachaohuang) [reply src/coprocessor/dag/expr/builtin_time.rs]: Is this just the problem of the test?

Comment (GinYM) [reply src/coprocessor/dag/expr/builtin_time.rs]: Yes, just the problem of test.

Comment (rleungx) [reply src/coprocessor/dag/expr/builtin_time.rs]: Do we need to test these invalid cases? @huachaohuang

Comment (huachaohuang) [reply src/coprocessor/dag/expr/builtin_time.rs]: I think it's fine. @GinYM if you are interested, please help to improve these DateTime tests in another PR :)

Comment (GinYM) [reply src/coprocessor/dag/expr/builtin_time.rs]: @huachaohuang Sure, I will try it.

**Pull request merged**